src/platform/input/sdl2mouse.cpp

```diff
@@ -0,0 +1,186 @@
+{
+    if (s_loadedCursors[load_index][load_direction] != NULL) {
+        // Cursor at this position already loaded. Ignoring subsequent load.
+        return;
+    }
+    // TODO: this should parse the ANI file and load the cursor correctly
+    SDL_Surface *surface = SDL_LoadBMP(file_location);
+    if (!surface) {
+        // Failed to load surface
+        return;
+    }
+    SDL_Cursor *cursor = SDL_CreateColorCursor(surface, 0, 0);
+    if (!cursor) {
+        // Failed to create cursor
+        return;
+    }
+    s_loadedCursors[load_index][load_direction] = cursor;
+}
+
+void SDL2Mouse::Init_Cursor_Resources()
+{
+    // First cursor is not loaded. As this is CURSOR_NONE
+    for (auto i = 1; i < CURSOR_COUNT; ++i) {
+        auto &cursor_info = m_cursorInfo[i];
+        if (cursor_info.texture_name.Is_Empty()) {
+            continue;
+        }
+
+        if (cursor_info.directions <= 1) {
+            char buffer[256]{};
+            snprintf(buffer, 256, "Data/Cursors/%s.ani", cursor_info.texture_name.Str());
+            Load_Cursor(buffer, i, 0);
+        } else {
+            for (auto direction = 0; direction < cursor_info.directions; direction++) {
+                char buffer[256]{};
+                snprintf(buffer, 256, "Data/Cursors/%s%d.ani", cursor_info.texture_name.Str(), direction);
+                Load_Cursor(buffer, i, direction);
+            }
+        }
+    }
+}
+
+void SDL2Mouse::Set_Cursor(MouseCursor cursor)
+{
+    Set_Mouse_Text(cursor);
+    m_currentCursor = cursor;
+}
+
+void SDL2Mouse::Set_Visibility(bool visibility)
+{
+    m_visible = visibility;
+    SDL_ShowCursor(visibility);
+}
+
+void SDL2Mouse::Add_SDL2_Event(SDL_Event *ev)
+{
+    auto &event = m_eventBuffer[m_nextFreeIndex];
+
+    // If no space (0 type == empty) for more events throw away event
+    if (event.type != 0) {
+        return;
+    }
+
+    event = *ev;
+
+    m_nextFreeIndex++;
+    if (m_nextFreeIndex >= MAX_EVENTS) {
+        m_nextFreeIndex = 0;
+    }
+}
+
+uint8_t SDL2Mouse::Get_Mouse_Event(MouseIO *io, int8_t unk)
+{
+    if (m_eventBuffer[m_nextGetIndex].type == 0) {
+        return false;
+    }
+
+    Translate_Event(m_nextGetIndex, io);
+
+    // Clear the current event - can be overwritten
+    m_eventBuffer[m_nextGetIndex].type = 0;
+    m_nextGetIndex++;
+    if (m_nextGetIndex >= MAX_EVENTS) {
+        m_nextGetIndex = 0;
+    }
+    return true;
+}
+
+void SDL2Mouse::Translate_Event(uint32_t message_num, MouseIO *io)
+{
+    auto &event = m_eventBuffer[message_num];
+
+    uint32_t frame_num = 1;
+    if (g_theGameClient != nullptr) {
+        frame_num = g_theGameClient->Get_Frame();
+    }
+
+    io->right_state = 0;
+    io->middle_state = 0;
+    io->left_state = 0;
+    io->right_frame = 0;
+    io->middle_frame = 0;
+    io->left_frame = 0;
+    io->wheel_delta = 0;
+    io->pos.y = 0;
+    io->pos.x = 0;
+
+    // TODO: mousewheel handling is a bit difficult - how to calculate a delta similar to win32mouse?
```

Comment:
Also double clicks and middle mouse button handling is missing here.

Reply:
Handled now